pkg/file/joiner/joiner.go

```diff
@@ -42,7 +54,17 @@ func (s *simpleJoiner) Size(ctx context.Context, address swarm.Address) (dataSiz
 		return 0, fmt.Errorf("invalid chunk content of %d bytes", chunkLength)
 	}
 
-	dataLength := binary.LittleEndian.Uint64(rootChunk.Data())
+	var chunkData []byte
+	if toDecrypt {
+		originalData, err := internal.DecryptChunkData(rootChunk.Data(), key)
+		if err != nil {
+			return 0, err
+		}
+		chunkData = originalData
+	} else {
+		chunkData = rootChunk.Data()
+	}
```

Comment:
```suggestion
	chunkData := rootChunk.Data()
	if toDecrypt {
		decryptedData, err := internal.DecryptChunkData(chunkData, key)
		if err != nil {
			return 0, err
		}
		chunkData = decryptedData
	}
```

ftr: i would also prefer to have it this way in `Join`, but it is not a blocker.